resources/sdk/purecloudkotlin/templates/ApiClient.kt.mustache

```diff
@@ -389,7 +446,8 @@ class ApiClient private constructor(builder: Builder) : AutoCloseable {
         val statusCode = response!!.statusCode
         val reasonPhrase = response.statusReasonPhrase
         val headers = response.getHeaders()
-        return if (statusCode in 200..299) {
+        // If the response indicates that the access_token has expired, and we wish to refresh it transparently, we'll return from this method and let the caller handle it
+        return if (statusCode in 200..299 || shouldRefreshAccessToken && statusCode == 401) {
```

Comment:
Why is the 401 response being handled as a success here? Same concerns as this condition mentioned in another comment.